virtual/vbox/vbox.go

```diff
@@ -55,9 +55,22 @@ func (err *VBoxErr) Error() string {
 type VM interface {
 	virtual.Instance
 	Snapshot(string) error
-	LinkedClone(context.Context, string, ...VMOpt) (VM, error)
+	LinkedClone(context.Context, string, ...VMOpt) (VM, error) // what is LinkedClone ?
```

Comment:
See [this link](https://docs.oracle.com/cd/E97728_01/E97727/html/vboxmanage-clonevm.html) of VirtualBox documentation. A linked clone is a clone of an existing VM whose disk is defined by its diff against the original VM (if I recall correctly). This is opposed to a full clone, whose disk is a full copy of the original one. So disk (and speed)wise, it is much more efficient to create linked clones than full clones.

Reply:
👍🏻 I will check it.

Reply:
So there exists always a "master" or "parent" VM, from which we create clones. You can see this on `sec02` when you run the following:
```
ntpd@sec02:~$ vboxmanage list vms
...
"kali{ecc3ea71}" {c8a577db-b6e7-4b9a-b79d-cd4fed8edc08}
...
"40541ce8f1e54c31940406462144e906" {6b2045aa-74ec-4005-a609-2da467add676}
"754a397f117d455cb3232a90685f1611" {a09c717f-2700-4969-92e6-d05ce662da8f}
"941ebe8faac142d68dccb9521de0d32e" {2e0bad87-126a-43be-878d-aa8f68f884ff}
"0868359c8e314aeb9b9a355657a79e64" {398ff831-e10e-4312-990b-c2b34c01a92f}
"1cc94f0b7650468ab7a42913e5e61e25" {1c1bf14a-03ce-4af4-b1dc-fa0a5d985d38}
...
```
Here, `kali{ecc3ea71}` is the "master" or "parent" VM, and all the other ones are linked clones of that one.